backend/private-graph/graph/schema.resolvers.go

```diff
@@ -4612,16 +4612,16 @@ func (r *queryResolver) ClientIntegration(ctx context.Context, projectID int) (*
 	}
 
 	firstSession := model.Session{}
-	err := r.DB.Model(&model.Session{}).Where("project_id = ?", projectID).Limit(1).Find(&firstSession).Error
-	if e.Is(err, gorm.ErrRecordNotFound) {
-		return integration, nil
-	}
+	err := r.DB.Model(&model.Session{}).Where("project_id = ?", projectID).Where("created_at > ?", time.Now().Add(time.Hour*24*-30)).Limit(1).Find(&firstSession).Error
```

Comment:
@et ended up adding this additional `where` clause to ensure we don't query sessions or errors outside the retention period. That's the reason the session/error wasn't loading for you. I also updated the CTA text to say "View an error" and "View a session" since there's a chance it won't be the first.

Reply:
Instead of hardcoding the time frame, can we change to actually use the retention date function:

https://github.com/highlight/highlight/blob/2614e6605eee217754bca2cb8ecf4e4f3ee47f1c/backend/private-graph/graph/schema.resolvers.go#L3773-L3782

Reply:
Yeah, I thought about doing that, but this query could potentially get hit so frequently I figured I'd avoid the extra query or join and just hard-code the 30 days.

Reply:
@ccschmitz and I chatted about this over slack, particularly if this was going to be too brittle (e.g. 60 day retention but last session was 31 days ago). But in that scenario, we should consider setup as not integrated and advise the user to check their setup.